Issue/11323 new product image picker

JorgeMucientes opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #11323 
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
Adds a new product image picker within Blaze campaign creation flow. 
- A new item is available in the media picker dialog displayed when editing Ad image
- A new fragment is loaded when selecting the option `Choose existing product photo`

This PR doesn't add any unit tests because there's barely any logic. It is mostly a simple UI and handling navigation. 

### Testing information
1. Trigger Blaze campaign creation flow
2. Select a product that doesn't have any image
3. Tap `Edit Ad` > `Change image` > `Choose existing product photo`
4. Check empty case is handled. 

Repeat the above steps and select a product with multiple images.
1. Select any of the available images and check how the UI is updated accordingly
2. Click save and notice how the Ad image is updated

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->


https://github.com/user-attachments/assets/5e40d610-d27f-4548-9cdb-92b7340dc01d

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->323

JorgeMucientes committed:
Add new action to image picker dialog to select images for given product

JorgeMucientes committed:
Handle taps on product image picker

JorgeMucientes committed:
Display product images upon tapping in new option

JorgeMucientes committed:
Update product selected image

JorgeMucientes committed:
Update product images UI for adaptive grid

JorgeMucientes committed:
Add toolbar and title to product photo picker

JorgeMucientes committed:
Handle empty case for product image picker

JorgeMucientes committed:
Fix detekt indentation issues

JorgeMucientes committed:
Fix title of photo picker

dangermattic commented:
<!--
  0 Errors
  1 Warning: Class <code>ProductImagePicker...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>ProductImagePickerViewModel</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

JorgeMucientes committed:
Update release notes

JorgeMucientes committed:
Merge branch 'refs/heads/trunk' into issue/11323-new-product-image-picker

JorgeMucientes committed:
Update release notes

JorgeMucientes committed:
Remove compose code duplication

JorgeMucientes commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/ad/ProductImagePickerFragment.kt:
```diff
@@ -0,0 +1,45 @@
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class ProductImagePickerFragment : BaseFragment() {
```
Thought of adding this new fragment in `products.images` package, but i think that would be more confusing as there's already `ProductImagesFragment` and related classes to handle product images. While this feature might be handy in the future, currently it is very specific to the Blaze feature. For those reasons, I'm keeping it inside the `blaze` package.

irfano commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/ad/ProductImageScreen.kt:
```diff
@@ -0,0 +1,121 @@
+
+@Composable
+fun ProductImagePickerScreen(
+    viewState: ProductImagePickerViewModel.ViewState,
```
- We referenced `ProductImagePickerViewModel` here, but it can be removed since we imported `com.woocommerce.android.ui.blaze.creation.ad.ProductImagePickerViewModel.ViewState`. I suggest removing `import com.woocommerce.android.ui.blaze.creation.ad.ProductImagePickerViewModel.ViewState` and also referencing  `ProductImagePickerViewModel` in `ProductImageGrid` for consistency.
- Also we can remove `Product.`

![image](https://github.com/user-attachments/assets/908153bc-e6e7-40a3-869e-2ada636c0b81)

irfano commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/ad/ProductImageScreen.kt:
```diff
@@ -0,0 +1,121 @@
+                fallback = painterResource(R.drawable.blaze_campaign_product_placeholder),
+                placeholder = painterResource(R.drawable.blaze_campaign_product_placeholder),
+                error = painterResource(R.drawable.blaze_campaign_product_placeholder),
+                contentDescription = "",
```
I tested selecting images with Talkback is enabled. When I select an image, Talkback says "unlabeled button." I suggest an improvement here. I tested the Google Photos app, and it says "photo taken on $date". Can we implement the same here. Feel free to open a separate issue if you prefer.

irfano commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/ad/ProductImagePickerViewModel.kt:
```diff
@@ -0,0 +1,62 @@
+        }
+    }
+
+    fun onImageSelected(productImage: Product.Image) {
```
np: We can remove `Product.` here.

JorgeMucientes committed:
Fix how ProductImagePickerViewModel and Product.Images are imported

JorgeMucientes committed:
Add non empty content description to product photo

JorgeMucientes commented:
Hey @irfano thanks for the feedback!! I've addressed the import issues. A bit differently, though, I left the fully qualified names so it's easier to check what type of object it's being issued. For example, instead of `Image`, I added `Product.Image`. Same for the `viewState`. 

As per the talkback service, good catch. However, I'm not sure how useful would it be to add the "photo taken on $date" in this case. I mean, it doesn't seem like the date is very relevant data in this context to help identify which picture is being selec4ed. I've added a short content description to improve the experience a bit.

Let me know if there's anything else that should be addressed before this PR can be approved :)

irfano reviewed (approved):
It's better now. Thank you for changes. LGTM! 👍🏻
